pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -326,6 +327,89 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Removes a token from the Pool's composition, withdrawing all funds from the Vault and sending them to
+     * `recipient`, and adjusting the weights of all other tokens.
+     *
+     * Tokens can only be removed if the Pool has more than 2 tokens, as it can never have fewer than 2. Token removal
+     * is also forbidden during a weight change, or if one is scheduled to happen in the future.
+     *
+     * The caller may aditionally pass a non-zero `burnAmount` to have some of their BPT be burned, which might be
+     * useful in some scenarios to account for the fact that the Pool now has fewer tokens.
+     */
+    function removeToken(
+        IERC20 token,
+        address recipient,
+        uint256 burnAmount
+    ) external authenticate nonReentrant whenNotPaused {
```

Comment:
As mentioned in the previous comment, we don't have any guarantees on how much of `token` we'll receive. We're then committing to burn `burnAmount` BPT of value in return for a substantially reduced amount of `token`.

It seems like we should allow the pool owner to provide minimum amount of `token` they would accept receiving to avoid losing value to MEV.

Reply:
Granted, this could be done in a pool manager contract but performing this in the pool is a one-liner.

Reply:
Ok; will put it in

Reply:
I spoke with nico about this and his feelings are that with https://github.com/balancer-labs/balancer-v2-monorepo/pull/1108#discussion_r857835184 we can just leave it to the pool manager contract to handle this.

Reply:
My intention was to add a redundant check that would guarantee the entire balance is removed from the Pool, and not a mitigation against MEV (since we're unable to do such a thing at this layer). However, my extra check is actually unnecessary since `deregisterTokens` would simply fail if the balance was not zeroed out.

Given that we have the `minimumAmountsOut` array already, I think it makes sense to add an argument and handle it here, as Jeff did. It is a bit more consistent with how `exitPool` works as well. However, let's please avoid having multiple people performing edits on the same branch: it makes it extremely annoying to update them.

Reply:
(also new features need tests as well, not just contract code)